src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -499,25 +498,38 @@ internal virtual void Validate()
 
         internal ApplicationConfiguration BuildConfiguration()
         {
+            ResolveAuthority();
             Validate();
             return Config;
         }
 
         #region Authority
-        private void CreateAuthorityInfoFromEnums()
+        private void ResolveAuthority()
         {
             if (Config.AuthorityInfo != null)
             {
-                return;
+                if (!string.IsNullOrEmpty(Config.TenantId) &&
```

Comment:
This looks like the same changes made in this PR here?
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2666/files

Should this be here?

Reply:
+1